src/client/gl-base-technique.cc

```diff
@@ -0,0 +1,56 @@
+#include "client/gl-base-technique.h"
+
+#include "client/atomic-command-queue.h"
+#include "core/logger.h"
+
+namespace zen::remote::client {
+
+GlBaseTechnique::GlBaseTechnique(
+    uint64_t id, AtomicCommandQueue* update_rendering_queue)
+    : id_(id), update_rendering_queue_(update_rendering_queue)
+{
+}
+
+GlBaseTechnique::~GlBaseTechnique() {}
+
+void
+GlBaseTechnique::Commit()
+{
+  if (pending_.data_damaged == false) return;
+
+  auto command = CreateCommand(
+      [this]() { rendering_.render_mode = pending_.render_mode; });
```

Comment:
We need to copy pending_.render_mode here.
This prevent the copied render_mode from being modified between when committed and when the command is executed.

```suggestion
  auto command = CreateCommand(
      [rendering_mode = pending_.rendering_mode, this]() { rendering_.render_mode = pending_.render_mode; });
```